app/models/user.rb

```diff
@@ -1,6 +1,9 @@
 class User < ActiveRecord::Base
   has_secure_password
 
+  has_many :user_group_memberships
+  has_many :groups, through: :user_group_memberships
```

Comment:
From the DB perspective a join table is the right name. The difference is on AR's side, where one of the relationships is manifested as a model and the other one doesn't.  :older_man: 

The name is great! I think it's good to be expressive. At the same time though, it's good to double check where something can be reutilized. For example, we might have different memberships between User - Group and User - Event. Those memberships might be identical as far as the info that they represent. In that case I would use a general `Membership` model, but if the memberships tend to be different then your route is the way to go. :smile: